bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/JournalChannel.java

```diff
@@ -177,7 +190,21 @@ private JournalChannel(File journalDirectory, long logId,
             }
             fc = channel.getFileChannel();
             formatVersion = formatVersionToWrite;
+            bc = bcBuilder.create(fc, writeBufferSize);
+        } else {  // open an existing file
+            fc = channel.getFileChannel();
+            formatVersion = formatVersionToWrite;
+        }
+        if (fRemoveFromPageCache) {
+            this.fd = NativeIO.getSysFileDescriptor(channel.getFD());
+        } else {
+            this.fd = -1;
+        }
+    }
 
+    public void writeHeader() throws IOException {
+        try {
+            bc = bcBuilder.create(fc, writeBufferSize);
```

Comment:
the `bc` is already created at line 193, right? if so, I think this line is redundant.